lumicks/pylake/kymo.py

```diff
@@ -248,6 +249,56 @@ def plot_rgb(self, **kwargs):
         image = image / np.max(image)
         return self._plot(image, **kwargs)
 
+    def _downsample_channel(self, n, xy, reduce=np.mean, where="center"):
+        force = self.file._get_force(n, xy)
+        # downsample exactly over the scanline time range
+        min_times = self.timestamps[0, :].astype(np.int64)
+        max_times = self.timestamps[-1, :].astype(np.int64)
+        time_ranges = [(mini, maxi) for mini, maxi in zip(min_times, max_times)]
+        return force.downsampled_over(time_ranges, reduce=reduce, where=where)
+
+    def plot_with_force(self, force_channel, color_channel, aspect_ratio=0.25,
+                        reduce=np.mean, where="center", **kwargs):
+        """Plot kymo with force channel downsampled over scan lines
+
+        Parameters
+        ----------
+        force_channel: str
+            name of force channel to downsample and plot
+        color_channel: str
+            color channel of kymo to plot ('red', 'green', 'blue', 'rgb')
+        aspect_ratio: float
+            axes aspect ratio
+        reduce : callable
+            The `numpy` function which is going to reduce multiple samples into one.
+            Forwarded to :func:`Slice.downsampled_over`
+        where : str
+            Where to put the final time point.
+            Forwarded to :func:`Slice.downsampled_over`
+        **kwargs
+            Forwarded to :func:`Slice.plot`.
+        """
+        def set_aspect_ratio(axis, ar):
+            """This function forces a specific aspect ratio, can be useful when aligning figures"""
+            axis.set_aspect(ar*np.diff(axis.get_xlim())[0] / np.abs(np.diff(axis.get_ylim()))[0])
+
+        import matplotlib.pyplot as plt
+
+        force = self._downsample_channel(force_channel[-2], force_channel[-1])
```

Comment:
`where` and `reduce` are not being propagated down.

Reply:
good catch ... fixed it for `reduce` and after some discussion with @JoepVanlier it doesn't make sense to give the `where` option to users since we are defining the line time at the central timestamp